frontend/common/src/components/WorkExperience/WorkExperience.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+import { Checkbox } from "../form/Checkbox";
+import { Input } from "../form/Input";
+
+export const WorkExperience = () => {
+  const rules = { required: true };
+  const intl = useIntl();
+
+  // given that the current role state needs to effect whether an end date is required according to the wireframe, here is something to handle that
+  // default state is required as box is unchecked by default, onChange in the toggle input calls the function to flip its state which is then called in the end date input
+  // adjusting required or not
+  const [checkboxState, setCheckbox] = useState({ required: true });
+  const checkboxToggle = () => {
+    if (checkboxState.required === true) {
+      setCheckbox({ required: false });
+    } else {
+      setCheckbox({ required: true });
+    }
+  };
```

Comment:
You're on the right track, but we use a library to handle form state for us, meaning we don't need to add any onChange handlers or set values on our inputs.
Instead of defining your own toggle handler like this, use the library's [useWatch](https://react-hook-form.com/api/usewatch) hook. In this case, you would use the hook somewhat like this:
```javascript
const isCurrent = useWatch({name: "current-role", defaultValue: false});
...
<Checkbox
  ...
  rules={isCurrent ? {} : rules}
/>
```
I'd also make the rendering of the end date contingent on whether isCurrent is true.

Reply:
understood